File: qwencoder-eval/chat/eval-dev-quality/testdata/golang/light/phoneNumber.go
package light

import "errors"

func extractDigits(dirtyNumber string) (cleanNumber string, err error) {
	for _, c := range dirtyNumber {
		if c == ' ' || c == '.' || c == '(' || c == ')' || c == '-' || c == '+' {
			// Remove spaces, dots, parentheses, hyphens and pluses.
			continue
		}
		if c == '-' || c == '@' || c == ':' || c == '!' {
			return "", errors.New("punctuations not permitted")
		}
		if c < '0' || c > '9' {
			return "", errors.New("letters not permitted")
		}
		cleanNumber += string(c)
	}

	return cleanNumber, nil
}

File: qwencoder-eval/instruct/eval-dev-quality/testdata/java/light/src/main/java/com/eval/PhoneNumber.java
package com.eval;

public class PhoneNumber {
  static String extractDigits(String dirtyNumber) {
    String cleanNumber = "";
    for (int i = 0; i < dirtyNumber.length(); i++) {
      char c = dirtyNumber.charAt(i);
      if (c == ' ' || c == '.' || c == '(' || c == ')' || c == '-' ||
          c == '+') {
        // Remove spaces, dots, parentheses, hyphens and pluses.
        continue;
      }
      if (c == '-' || c == '@' || c == ':' || c == '!') {
        throw new IllegalArgumentException("punctuations not permitted");
      }
      if (c < '0' || c > '9') {
        throw new IllegalArgumentException("letters not permitted");
      }
      cleanNumber += c;
    }

    return cleanNumber;
  }
}
